src/screens/Settings/SettingsPage.js

```diff
@@ -40,49 +41,83 @@ const styles = StyleSheet.create({
   },
   content: {
     padding: 20
+  },
+  button: {
+    backgroundColor: colors.darkGray,
+    marginTop: 20
   }
 });
 
 class SettingsPage extends Component {
   static navigationOptions = () => ({ title: "Settings" });
 
   async componentDidMount() {
-    await this.props.getUserToken();
+    this.willFocusSub = this.props.navigation.addListener(
+      "willFocus",
+      async payload => await this.handleWillFocus(payload)
+    );
```

Comment:
Could you help me understand what this code is doing?

Reply:
This adds a action subscription to the navigation that listens for when the user is about to navigate to this component. When the user navigates to Settings, it runs handleWillFocus, which checks whether a user login exists. If a login doesn't exist (i.e. a user isn't logged in), it navigates to the login form instead of Settings.